pkg/spi/plugin.go

```diff
@@ -0,0 +1,16 @@
+package spi
```

Comment:
As a nit -- would you mind rename `APISpec` to something like `Interface`?    `API` looks jarring when there are names like `APIsRequest` and `APIsResponse`.  

Also for the method `APIs()`, `Implements()` looks a little nicer to me.  It also matches how Docker engine does activation with the field `Implements`:  https://docs.docker.com/engine/extend/plugin_api/#/pluginactivate

So how about 
```
type Interface struct {
   Name string
   Version string
}

type Plugin interface {
    Implements() ([]Interface, error)
}
```
Then the RPC response can be named `HandshakeResponse` and it will look like
```
{
     "Implements" : [ ....]
     "Interfaces" : [  /* list of interface descriptions like example input / output via reflection */ ]
}
```